spinnaker/spinnaker_system/google_http_lb_test.py

```diff
@@ -0,0 +1,1256 @@
+      {
+        'cloudProvider': 'gce',
+        'provider': 'gce',
+        'stack': bindings['TEST_STACK'],
+        'credentials': bindings['GCE_CREDENTIALS'],
+        'region': bindings['TEST_GCE_REGION'],
+        'loadBalancerType': 'HTTP',
+        'loadBalancerName': self.__lb_name,
+        'portRange': '80',
+        'defaultService': {
+          'name': 'default-backend-service',
+          'backends': [],
+          'healthCheck': self.__default_hc,
+        },
+        'certificate': '',
+        'hostRules': [
+          {
+            'hostPatterns': ['host1.com', 'host2.com'],
+            'pathMatcher': {
+              'pathRules': [
+                {
+                  'paths': ['/path', '/path2/more'],
+                  'backendService': {
+                    'name': 'backend-service',
+                    'backends': [],
+                    'healthCheck': self.__default_hc,
+                  }
+                }
+              ],
+              'defaultService': {
+                'name': 'pm-backend-service',
+                'backends': [],
+                'healthCheck': self.__default_hc,
+              }
+            }
+          }
+        ],
+        'type': 'upsertLoadBalancer',
+        'availabilityZones': {bindings['TEST_GCE_REGION']: []},
+        'user': '[anonymous]'
+      }
+    ]
+    self.__default_delete = [
+      {
+        'type': 'deleteLoadBalancer',
+        'cloudProvider': 'gce',
+        'loadBalancerType': 'HTTP',
+        'loadBalancerName': self.__lb_name,
+        'region': bindings['TEST_GCE_REGION'],
+        'regions': [bindings['TEST_GCE_REGION']],
+        'credentials': bindings['GCE_CREDENTIALS'],
+        'user': '[anonymous]'
+      }
+    ]
+
+
+  def _make_bs_link(self, bs):
+    '''Make a fully-formatted backend service link.
+    '''
+    return (GCE_URL_PREFIX + self.bindings['GCE_PROJECT'] +
+            '/global/backendServices/' + bs)
+
+
+  def _make_hc_link(self, hc):
+    '''Make a fully-formatted health check link.
+    '''
+    return (GCE_URL_PREFIX + self.bindings['GCE_PROJECT'] +
+            '/global/httpHealthChecks/' + hc)
+
+
+  def upsert_full_load_balancer(self):
+    '''Upserts L7 LB with full hostRules, pathMatchers, etc.
+
+    Calls the upsertLoadBalancer operation with a payload, then verifies that
+    the expected resources are visible on GCP.
+    '''
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=self.__default_upsert,
+      descriptiopn='Upsert L7 Load Balancer: ' + self.__lb_name,
+      application=self.TEST_APP
+    )
+
+    full_hc_name = self.__default_hc['name']
+    hc_link = self._make_hc_link(self.__default_hc['name'])
+
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    (contract_builder.new_clause_builder('Health Checks Created',
+                                         retryable_for_secs=30)
+     .list_resource('httpHealthChecks')
+     .contains_pred_list(
+       [
```

Comment:
Here and below, I think often you want PathEqPredicate to look for exact values.
The full_hc_name is contains since it is a subset.
The port is an integer so Contains is the same as Eq.
PathContainsPredicate('requestPath', '/') just means it has a slash in it, not it is the root. Is that what you meant?

A lot of the clauses below are using what looks like exact string literals, so Eq is more precise to what you mean for those.